src/test/spec/unified_runner/matcher.rs

```diff
@@ -39,6 +48,145 @@ pub(crate) fn events_match(
     }
 }
 
+#[cfg(feature = "tracing-unstable")]
+pub(crate) fn tracing_events_match(
+    actual: &TracingEvent,
+    expected: &ExpectedMessage,
+) -> Result<(), String> {
+    if actual.target != expected.target {
+        return Err(format!(
+            "Expected and actual tracing event components do not match: expected {}, got {}",
+            expected.target, actual.target
+        ));
+    }
+    if actual.level != expected.level {
+        return Err(format!(
+            "Expected and actual tracing event levels do not match: expected {}. got {}",
+            expected.level, actual.level
+        ));
+    }
+
+    if let Some(failure_should_be_redacted) = expected.failure_is_redacted {
+        match actual.fields.get("failure") {
+            Some(failure) => {
+                match failure {
+                    TracingEventValue::String(failure_str) => {
+                        // lazy_static saves us having to recompile this regex every time this
+                        // function is called.
+                        lazy_static! {
```

Comment:
as an aside, I kind of assumed that a nested `lazy_static` like this would get re-evaluated on every function call, but this was explicitly recommended in the regex crate docs as a way to avoid re-compiling within helper methods that use regexes: https://docs.rs/regex/latest/regex/index.html#example-avoid-compiling-the-same-regex-in-a-loop

Reply:
added comments in the relevant places for command errors and I/O errors 👍